cpp/src/strings/substring.cu

```diff
@@ -257,30 +249,111 @@ struct dispatch_substring_from_fn {
             std::enable_if_t<not std::is_integral<PositionType>::value>* = nullptr>
-  std::unique_ptr<column> operator()(strings_column_view const&,
-                                     column_view const&,
-                                     column_view const&,
-                                     rmm::mr::device_memory_resource*,
-                                     cudaStream_t) const
+  std::unique_ptr<column> operator()(column_device_view const& d_column,
+                                     size_type null_count,
+                                     column_view const& starts_column,
+                                     column_view const& stops_column,
+                                     rmm::mr::device_memory_resource* mr,
+                                     cudaStream_t stream) const
   {
     CUDF_FAIL("Positions values must be an integral type.");
   }
 };
 
-template <>
-std::unique_ptr<column> dispatch_substring_from_fn::operator()<bool>(
-  strings_column_view const&,
-  column_view const&,
-  column_view const&,
-  rmm::mr::device_memory_resource*,
-  cudaStream_t) const
-{
-  CUDF_FAIL("Positions values must not be bool type.");
-}
+// When slice_strings is invoked with a delimiter string and a delimiter count, we need to
+// compute the start and end indices of the substring. This functor accomplishes that.
+struct compute_substring_indices {
+  template <typename DelimiterItrT>
+  void operator()(column_device_view const& d_column,
+                  DelimiterItrT const delim_itr,
+                  size_type delimiter_count,
+                  size_type* start_char_pos,
+                  size_type* end_char_pos,
+                  rmm::mr::device_memory_resource* mr,
+                  cudaStream_t stream) const
+  {
+    auto strings_count = d_column.size();
+
+    thrust::for_each_n(
+      rmm::exec_policy(stream)->on(stream),
+      thrust::make_counting_iterator<size_type>(0),
+      strings_count,
+      [delim_itr, delimiter_count, start_char_pos, end_char_pos, d_column] __device__(
+        size_type idx) {
+        auto const& delim_val_pair = delim_itr[idx];
+        auto const& delim_val      = delim_val_pair.first;  // Don't use it yet
 
+        // If the column value for this row is null, result is null.
+        // If the delimiter count is 0, result is empty string.
+        // If the global delimiter or the row specific delimiter is invalid or if it is empty, row
+        // value is empty.
+        if (!d_column.is_null(idx) && delim_val_pair.second && !delim_val.empty()) {
+          auto const& col_val = d_column.element<string_view>(idx);
+
+          // If the column value for the row is empty, the row value is empty.
+          if (!col_val.empty()) {
+            auto const col_val_len   = col_val.length();
+            auto const delimiter_len = delim_val.length();
+
+            auto nsearches      = (delimiter_count < 0) ? -delimiter_count : delimiter_count;
```

Comment:
Either add a comment on the meaning of this check, or add a variable to help make this easier to understand, something like
`bool const left_to_right_search = delimiter > 0;`

Reply:
done